### PR TITLE
January 2019 Update 

### LINKED ISSUE #59: Course Members page needs a redesign
The members page should either be:

Completely separate from /course/ (i.e. not in a tabbed box element)

or

Member info should come up as a modal box that can be dismissed.

It's annoying to look through member profiles because when you press back after looking at a profile, it reloads /course/ and you're taken to the course projects list.

### LINKED ISSUE #80: When professor creates a project, he is the TA and Scrum Master of the project.
can we set these fields to a null entity? Options...

### LINKED ISSUE #106: Need to apply User Role changes to Course view as well.
Currently edit_course throws error for TA's.

### LINKED ISSUE #109: TA's should be able to post Course Updates

### LINKED ISSUE #125: TSR Edit - Cancel Button
TSR Edit needs a cancel button. Also needs to be styled.

### LINKED ISSUE #148: Separate Instructors and TAs in Course Members tab
image

### LINKED ISSUE #149: New Feature: Disable ability to Request to Join project
Allow course owners to enable/disable students having the ability to request to join projects.

### LINKED ISSUE #156: Match Stats page doesn't have Breadcrumbs
Has 'Home' breadcrumb and thats it. Should be:

Home -> My Matches -> {Project Name} Match Stats

### LINKED ISSUE #157: Adjust notification dropdown
Currently the title 'Your Notifications' and your actual notifications look exactly the same.

image

### LINKED ISSUE #158: Adjust Account/Profile dropdown
Still doesn't look that great.

### LINKED ISSUE #169: Edit Resource
Add ability to edit a project resource

### PR DESCRIPTION
**Pull request by rmonroe**
**Tested**
 - Resolve #109 TA can't post course update
- Markdown Tooltips added to Bio and course description and changed to GitHub URL
- Multi-Course interest fix
- Change interest text to clarify Resubmit
- Resolve #158 Account drop-down rework 
- Resolve #59 and Resolve #148 Course member edit and split TAs and Profs
- Resolve #106 TA can edit Course
____________________________________________
**NEED TESTING**
- Added edit to project updates
- Fixes #156 Matchstats now has breadcrumbs
- Fixes #125 TSR cancel and re-design
- Professor can post resource for project
- Fixes #157 notification title fix
- Resolve #149 Add disable request to join
- Fixes #80 prof and ta no longer auto added as Scrum master and Ta
- Close #169 add ability to edit resource
- Add ability to delete a resource
- Resource no longer posted with UTC date
- Hides inactive dropdown if no inactive courses

### CODE REVIEW

File: teamwork/apps/courses/views/CourseView.py
Comment (Kapn) [inline teamwork/apps/courses/views/CourseView.py]: This entire section can just be:
``` 
if user_role == "student":
              create message
              redirect
```
We can delete the 4 extra lines of code. This is repeated in several views.

File: teamwork/apps/projects/migrations/0006_auto_20190125_2151.py
Comment (Kapn) [inline teamwork/apps/projects/migrations/0006_auto_20190125_2151.py]: Don't merge these into production until this PR gets the updates from the squash PR

File: teamwork/apps/projects/forms.py
Comment (Kapn) [inline teamwork/apps/projects/forms.py]: Need to fix the label text. "Do no allow"

File: teamwork/templates/core/matchstats.html
Comment (Kapn) [inline teamwork/templates/core/matchstats.html]: Breadcrumb for Matchstats should display: "Home > My Matches > Matchstats( with active css class)" currently doesn't show the current window.

File: teamwork/templates/core/view_matches.html
Comment (Kapn) [inline teamwork/templates/core/view_matches.html]: Breadcrumb for My Matches should display: "Home > My Matches (active)". Currently only displays "Home"

File: teamwork/templates/projects/add_resource.html
Comment (Kapn) [inline teamwork/templates/projects/add_resource.html:5]: Same as others: Breadcrumb should be: "Home > CourseName > ProjectName > Add Resource (active)"

File: teamwork/templates/projects/edit_project.html
Comment (Kapn) [inline teamwork/templates/projects/edit_project.html]: Lock request to join shouldn't be in the same div as the box-title.

image

File: teamwork/templates/projects/post_update.html
Comment (Kapn) [inline teamwork/templates/projects/post_update.html:5]: Same : "Home > Courses > CourseName > Post Update(active)

File: teamwork/templates/projects/tsr_member.html
Comment (Kapn) [inline teamwork/templates/projects/tsr_member.html:5]: Same should be "Home > CourseName > ProjectName > TSR Edit

File: teamwork/templates/projects/tsr_scrum_master.html
Comment (Kapn) [inline teamwork/templates/projects/tsr_scrum_master.html:5]: Same

File: teamwork/templates/projects/tsr_scrum_master.html
Comment (Kapn) [inline teamwork/templates/projects/tsr_scrum_master.html:57]: Form looks sweet

File: teamwork/templates/projects/update_post_update.html
Comment (Kapn) [inline teamwork/templates/projects/update_post_update.html:1]: Looks good. We should add a message to display after editing/deleting an 'update'

File: teamwork/templates/projects/view_project.html
Comment (Kapn) [inline teamwork/templates/projects/view_project.html]: Should we allow TA's to post project updates too?

**Review by Kapn — CHANGES_REQUESTED** [review]
Looks good, couple small tweaks needed in my opinion 💯

File: teamwork/apps/projects/migrations/0006_auto_20190125_2151.py
Comment (rmonroe) [reply teamwork/apps/projects/migrations/0006_auto_20190125_2151.py]: Done

File: teamwork/templates/projects/tsr_scrum_master.html
Comment (rmonroe) [reply teamwork/templates/projects/tsr_scrum_master.html:57]: Thanks

File: teamwork/apps/courses/views/CourseView.py
Comment (rmonroe) [reply teamwork/apps/courses/views/CourseView.py]: Fixed

File: teamwork/templates/projects/update_post_update.html
Comment (rmonroe) [reply teamwork/templates/projects/update_post_update.html:1]: Done for Resources as well

**Comment by rmonroe** [comment]
MERGE PR #211 FIRST, to add changes requested

**Comment by rmonroe** [comment]
Ensure you back up DB before, I had to run a DB flush after the migration Squash was merged in. 
Scrum master threw a NOT NULL constraint failed

**Review by Kapn — APPROVED** [review]

**Pull request merged**
